File: 0334-increasing-triplet-subsequence/0334-increasing-triplet-subsequence.cpp
class Solution {
public:
    bool increasingTriplet(vector<int>& nums) {
        int left = INT_MAX,mid = INT_MAX;
        for(auto it: nums){
            if(it <= left)
                left = it;
            else if(it <= mid)
                mid = it;
            
            else
                return true;
        }
        
        return false;
    }
};

File: 0334-increasing-triplet-subsequence/0334-increasing-triplet-subsequence.py
import sys
class Solution:
    def increasingTriplet(self, nums: List[int]) -> bool:
        c1 = sys.maxsize
        c2 = sys.maxsize
        for i in nums:
            if i <= c1:
                c1 = i
            elif i <= c2:
                c2 = i
            else:
                return True
        
        return False
        